plugins/modules/ntnx_volume_groups.py

```diff
@@ -0,0 +1,413 @@
+
+    disk_spec = dict(
+        size_gb=dict(type="int"),
+        storage_container=dict(
+            type="dict", options=entity_by_spec, mutually_exclusive=mutually_exclusive
+        ),
+    )
+
+    client_spec = dict(
+        uuid=dict(type="str"),
+        iscsi_iqn=dict(type="str"),
+        iscsi_ip=dict(type="str"),
+        client_password=dict(type="str", no_log=True),
+    )
+
+    module_args = dict(
+        name=dict(type="str"),
+        desc=dict(type="str"),
+        cluster=dict(
+            type="dict", options=entity_by_spec, mutually_exclusive=mutually_exclusive
+        ),
+        target_prefix=dict(type="str"),
+        volume_group_uuid=dict(type="str"),
+        flash_mode=dict(type="bool", default=False),
+        disks=dict(type="list", elements="dict", options=disk_spec),
+        vms=dict(
+            type="list",
+            elements="dict",
+            options=entity_by_spec,
+            mutually_exclusive=mutually_exclusive,
+        ),
+        load_balance=dict(type="bool", default=False),
+        clients=dict(
+            type="list",
+            elements="dict",
+            options=client_spec,
+            mutually_exclusive=[("uuid", "iscsi_iqn", "iscsi_ip")],
+        ),
+        CHAP_auth=dict(type="bool", default=False),
+        target_password=dict(type="str", no_log=True),
+    )
+
+    return module_args
+
+
+def create_volume_group(module, result):
+    volume_group = VolumeGroup(module)
+    spec, error = volume_group.get_spec()
+    if error:
+        result["error"] = error
+        module.fail_json(msg="Failed generating volume_groups spec", **result)
+
+    if module.check_mode:
+        result["response"] = spec
+        result["response"]["disks"] = module.params.get("disks")
+        result["response"]["vms"] = module.params.get("vms")
+        result["response"]["clients"] = module.params.get("clients")
+        return
+
+    resp = volume_group.create(spec)
+    task_uuid = resp["data"]["extId"][-36:]
+    result["changed"] = True
+    result["response"] = resp
+    result["task_uuid"] = task_uuid
+    resp = wait_for_task_completion(module, result)
+    volume_group_uuid = resp["entity_reference_list"][0]["uuid"]
+    result["volume_group_uuid"] = volume_group_uuid
+    resp = volume_group.read(volume_group_uuid)
+    result["response"] = resp.get("data")
+
+    # create disks
+    if module.params.get("disks"):
+        for disk in module.params["disks"]:
+            spec, err = VDisks.get_spec(module, disk)
+            if err:
+                result["warning"] = "Disk is not created. Error: {0}".format(err)
+                result["skipped"] = True
+                continue
+
+            vdisk_resp = volume_group.create_vdisk(spec, volume_group_uuid)
+
+            task_uuid = vdisk_resp["task_uuid"]
+            wait_for_task_completion(module, {"task_uuid": task_uuid})
+
+        disks = volume_group.read(volume_group_uuid, endpoint="/disks")
```

Comment:
Let the library code decide the endpoint rather than client code. Easy to update in future by directly changing library code